cloudformation/template.yaml

```diff
@@ -1285,12 +1271,42 @@ Resources:
       Layers:
         - !Ref LambdaCommonLayer
 
+  CognitoUserMigrationTriggerFn:
+    Type: AWS::Serverless::Function
+    Condition: HasLegacyUserPool
+    Properties:
+      FunctionName: !Sub '${AWS::StackName}-CognitoUserMigrationTriggerFn'
+      CodeUri: ../lambdas/cognito-user-migration-trigger
+      Handler: index.handler
+      MemorySize: 128
+      Role: !GetAtt CognitoUserMigrationTriggerExecutionRole.Arn
+      Runtime: nodejs12.x
+      Timeout: 3
+
+  # Adjust this and the next pool accordingly any time you make a breaking change to pools.
   CognitoUserPool:
     Type: AWS::Cognito::UserPool
     DeletionPolicy: Retain
     UpdateReplacePolicy: Retain
+    Condition: HasLegacyUserPool
     Properties:
       UserPoolName: !Ref CognitoIdentityPoolName
+      Policies:
+        PasswordPolicy:
+          MinimumLength: 12
+          RequireLowercase: true
+          RequireNumbers: true
+      Schema:
+        - AttributeDataType: String
+          Name: email
+          Required: false
+  
+  CognitoUserPool0:
```

Comment:
Same as above. Can we use names instead of numbers for the separate pools?

Reply:
Numbers are used so it's easier to simply transition between them, in case we have to enact breaking changes to them in the future. Think of it as an ABI versioning scheme so we aren't running into issues with Cognito's immutability.

We can find better long-term solutions to this later (like delegating it all to a custom resource like we already do with the edge lambda).